Migrate the debugging guide to RustRover (#1800) (#1802)
tauri-apps/tauri-docs: i18n/fr/docusaurus-plugin-content-docs/current/guides/debugging/rustrover.md

# Debugging in RustRover

In this guide, we’ll be setting up JetBrains RustRover for debugging the [Core Process of your Tauri app].

## Setting up a Cargo project 

Depending on which frontend stack is used in a project, the project directory may or may not be a Cargo project. By default, Tauri places the Rust project in a subdirectory called `src-tauri`. It creates a Cargo project in the root directory only if Rust is used for frontend development as well.

> If there's no the `Cargo.toml` file at the top level, you need to attach the project manually. Open the Cargo tool window (in the main menu, go to **View | Tool Windows | Cargo**), click **+** (**Attach Cargo Project**) on the toolbar, and select the `src-tauri/Cargo.toml` file.

Reviewer: ```suggestion
If there's no `Cargo.toml` file at the top level, you need to attach the project manually. Open the Cargo tool window (in the main menu, go to **View | Tool Windows | Cargo**), click **+** (**Attach Cargo Project**) on the toolbar, and select the `src-tauri/Cargo.toml` file.
```

![Rename Configuration](/img/guides/debugging/rustrover/rename-configuration-light.png#gh-light-mode-only)
![Rename Configuration](/img/guides/debugging/rustrover/rename-configuration-dark.png#gh-dark-mode-only)

### Development Server

The above configuration will use Cargo directly to build the Rust application and attach the debugger to it. This means we completely sidestep the Tauri CLI, so features like the `beforeDevCommand` and `beforeBuildCommand` will **not** be executed. We need to take care of that by running the development server manually.

To create the corresponding Run configuration, you need to check the actual development server in use. Look for the `src-tauri/tauri.conf.json` file and find the following line:

```json
    "beforeDevCommand": "pnpm dev"
```

For `npm`, `pnpm`, or `yarn`, you could use the **npm** Run Configuration, for example:

![NPM Configuration](/img/guides/debugging/rustrover/npm-configuration-light.png#gh-light-mode-only)
![NPM Configuration](/img/guides/debugging/rustrover/npm-configuration-dark.png#gh-dark-mode-only)

Make sure you have the correct values in the **Command**, **Scripts**, and **Package Manager** fields.

> If your development server is `trunk` meaning that your frontend framework uses WebAssembly, you could use the generic **Shell Script** Run Configuration:

Reviewer: ```suggestion
If your development server is `trunk` for Rust-based WebAssembly frontend frameworks, you could use the generic **Shell Script** Run Configuration:
```
